src/planwise/component/engine.clj

```diff
@@ -99,19 +101,97 @@
+    source))
+
+(defn compute-initial-scenario-by-point
+  [engine project]
+  (println "compute-initial-scenario-by-point")
+  (let [provider-set-id  (:provider-set-id project)
+        providers        (project-providers engine project) ;sort by capacity
+        sources          (sources-set/list-sources-in-set (:sources-set engine) (:source-set-id project))
+        algorithm        (:coverage-algorithm project)
+        filter-options   (get-in project [:config :coverage :filter-options])
+        fn-sources-under (fn [provider] (sources-set/list-sources-under-provider-coverage (:sources-set engine) (:source-set-id project) (:id provider) algorithm filter-options))
+        fn-select-by-id  (fn [sources ids] (filter (fn [source] (ids (:id source))) sources))
+        result-step1     (reduce ; over providers
+                          (fn [computed-state provider]
+                            (let [providers                 (:providers computed-state)
+                                  sources                   (:sources computed-state)
+                                  id-sources-under-coverage (set (map :id (fn-sources-under provider)))         ; create set with sources' id
+                                  sources-under-coverage    (fn-select-by-id sources id-sources-under-coverage) ; updated sources under coverage
+                                  total-demand              (sum-up sources-under-coverage :quantity)        ; total demand requested to current provider
+                                  updated-sources           (map (fn [source] (update-source-if-needed source id-sources-under-coverage provider total-demand)) sources)]
+                              {:providers (conj providers (assoc provider :satisfied (min (:capacity provider) total-demand)))
+                               :sources updated-sources}))
+                          {:providers nil
+                           :sources sources}
+                          providers)
+        result-step2     (map (fn [provider]  ; resolve unsatisfied demand per provider
+                                (let [sources                   (:sources result-step1)
+                                      id-sources-under-coverage (set (map :id (fn-sources-under provider)))
+                                      sources-under-coverage    (fn-select-by-id sources id-sources-under-coverage) ; updated sources under coverage
+                                      total-demand              (sum-up sources-under-coverage :quantity)]
+                                  (assoc provider :unsatisfied total-demand)))
+                              (:providers result-step1))]
+
+    (let [updated-sources           (map (fn [source] ; persist only id and quantity
+                                           {:id (:id source)
+                                            :quantity (:quantity source)})
+                                         (:sources result-step1))
```

Comment:
This can also be done with `(map #(select-keys % [:id :quantity]) (:sources result-step1))`